djangocms_versioning/admin.py

```diff
@@ -537,6 +537,19 @@ def _get_unpublish_link(self, obj, request, disabled=False):
             {"unpublish_url": unpublish_url, "disabled": disabled},
         )
 
+    def _get_published_link(self, obj, request):
+        """Helper function to get the link to the published page"""
+        if not obj.state == PUBLISHED:
+            # Don't display the link if it isn't published
+            return ""
+
+        published_url = obj.content.get_absolute_url()
```

Comment:
I'd suggest to change this to `getattr(obj.content, "get_absolute_url", None)` and assume that not all objects have an absolute url. Then `if not published_url: return ""` to make sure the eye is only shown if there is an url.